pkgs/sdk/server-ai/src/LdAiClient.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.Json;
+using LaunchDarkly.Sdk.Server.Ai.Adapters;
+using LaunchDarkly.Sdk.Server.Ai.Config;
+using LaunchDarkly.Sdk.Server.Ai.DataModel;
+using LaunchDarkly.Sdk.Server.Ai.Interfaces;
+using LaunchDarkly.Sdk.Server.Interfaces;
+using Mustache;
+
+namespace LaunchDarkly.Sdk.Server.Ai;
+
+/// <summary>
+/// The LaunchDarkly AI client. The client is capable of retrieving AI configurations from LaunchDarkly,
+/// and generating events specific to usage of the AI configuration when interacting with model providers.
+/// </summary>
+public sealed class LdAiClient
+{
+    private readonly ILaunchDarklyClient _client;
+    private readonly ILogger _logger;
+
+    /// <summary>
+    /// Constructs a new LaunchDarkly AI client.
+    ///
+    /// Example:
+    /// <code>
+    /// var config = Configuration.Builder("my-sdk-key").Build();
+    /// var client = new LdClient(config);
+    /// var aiClient = new LdAiClient(client);
+    /// </code>
+    ///
+    /// </summary>
+    /// <param name="client">a LaunchDarkly Server-side SDK client instance</param>
+    public LdAiClient(ILaunchDarklyClient client)
+    {
+        _client = client ?? throw new ArgumentNullException(nameof(client));
+        _logger = _client.GetLogger();
+    }
+
+
+    // This is the special Mustache variable that can be used in prompts to access the current
+    // LaunchDarkly context. For example, {{ ldctx.key }} will return the context key.
+    private const string LdContextVariable = "ldctx";
+
+    /// <summary>
+    /// Retrieves a LaunchDarkly AI config identified by the given key. The return value
+    /// is an <see cref="LdAiConfigTracker"/>, which makes the configuration available and
+    /// provides convenience methods for generating events related to model usage.
+    ///
+    /// Any variables provided will be interpolated into the prompt's messages.
+    /// Additionally, the current LaunchDarkly context will be available as 'ldctx' within
+    /// a prompt message.
+    ///
+    /// </summary>
+    /// <param name="key">the flag key</param>
+    /// <param name="context">the context</param>
+    /// <param name="defaultValue">the default config, if unable to retrieve from LaunchDarkly</param>
+    /// <param name="variables">the list of variables used when interpolating the prompt</param>
+    /// <returns>an AI config tracker</returns>
+    public LdAiConfigTracker ModelConfig(string key, Context context, LdAiConfig defaultValue,
+        IReadOnlyDictionary<string, object> variables = null)
+    {
+
+        var result = _client.JsonVariation(key, context, LdValue.Null);
+
+        var defaultTracker = new LdAiConfigTracker(_client, key, defaultValue, context);
```

Comment:
I was previously using a detailed evaluation here, which had the ability to directly confirm if the default was returned. This was nice. 

Using a normal eval, I can only detect that by passing in a particular default and confirming that is what came back out. I could pass in some sentinel value that is highly unlikely to be anything else.

This is in service of not having to serialize the passed in `LdAiConfig` and pass it into the SDK.